core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -684,33 +666,14 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
     }
   }
 
-  /**
-   * Gets package from datastore and confirms it is not a binding.
-   */
-  private def pkgLookup(pkg: FullyQualifiedEntityName)(implicit transid: TransactionId): Future[WhiskPackage] = {
-    getPackage(pkg).filter {
-      _.binding.isEmpty
-    } recoverWith {
-      case _: ArtifactStoreException | DeserializationException(_, _, _) =>
-        // if the package lookup fails or the package doesn't conform to expected invariants,
-        // fail the request with BadRequest so as not to leak information about the existence
-        // of packages that are otherwise private
-        logging.debug(this, s"package which does not exist")
-        Future.failed(RejectRequest(NotFound))
```

Comment:
we should add some tests for `resolveActionAndMergeParameters` to preserve the behavior represented by these request rejections (missing package or malformed package).